### ISSUE TITLE
Should we replace JOGL?

### ISSUE THREAD

**Issue by andreas-eberle** [jsettlers/settlers-remake#546]
I think we should consider switching away from JOGL to another OpenGL framework. This is because of:

* JOGL is dead. There has not been a release / any significant changes since 2015
* JOGL seems to be instable on some hardware / software configurations. E.g. see #274 and I also have a computer where it is crashing quite often at startup with an EXCEPTION_ACCESS_VIOLATION ). 
* As there won't be any updates for JOGL, it will be hard to maintain stability in the future.

Therefore I think we should consider switching from JOGL to another framework. Does anybody know a good lightweight framework? In a first research I only found [LWJGL](https://www.lwjgl.org/). 

@michaelzangl: What do you think? How much effort would it probably take to switch the framework?

**Comment by michaelzangl** [comment]
I suggested a switch to LWGL some time back. It seems to be pretty active.

Switching is not that difficult. The hardest part - which I did not trace yet - is to do the font rendering.

We can re-use the sound player. The other classes in go.graphics.swing need to be re-written. All in all, we have ~5 Classes (~1000LOC) to replace, most of which are just interface implementations.

We then need to replace the main window creation in jsettlers.main.swing with the new one.

I'd like to keep JOGL for now. While replacing the main game window is pretty easy, replacing the map creator window might be difficult. I have no idea how good `AWTGLCanvas` performs.

We could use GLG2D if we have problems with the rendering. The GLG2D project is not actively maintained, but they at accept merge requests after some months delay. And it uses JOGL and needs to be adjusted to use LWJGL.
